onadata/libs/serializers/xform_serializer.py

```diff
@@ -142,12 +152,33 @@ def get_enketo_url(self, obj):
 
             url = self._get_metadata(obj, 'enketo_url')
             if url is None:
-                url = _create_enketo_url(self.context.get('request'), obj)
+                enketo_urls = _create_enketo_urls(
+                    self.context.get('request'), obj)
+                url = enketo_urls.get('offline_url')
 
             return _set_cache(ENKETO_URL_CACHE, url, obj)
 
         return None
 
+    def get_enketo_single_submit_url(self, obj):
+        if obj:
+            _enketo_single_submit_url = cache.get(
+                '{}{}'.format(ENKETO_SINGLE_SUBMIT_URL_CACHE,
+                              obj.pk))
+            if _enketo_single_submit_url:
+                return _enketo_single_submit_url
+
+            url = self._get_metadata(obj, 'enketo_url')
+            if url is None:
+                enketo_urls = _create_enketo_urls(
+                    self.context.get('request'), obj)
+                url = enketo_urls.get('offline_url')
```

Comment:
Should we be saving this url to a MetaData object like here https://github.com/onaio/onadata/pull/1857/files#diff-9df83887f6e59d1b4fddcf131e8ae2afR198

Reply:
_If possible it would be awesome if all of the MetaData object creation for these URLs happened in one function... Would make this easier to track and modify in the future_

Reply:
Aaaaah..i'll add that.